K2Bridge/Visitors/DateRangeAggregationExpressionVisitor.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+namespace K2Bridge.Visitors
+{
+    using System;
+    using System.Text.RegularExpressions;
+    using K2Bridge.Models.Request;
+    using K2Bridge.Models.Request.Aggregations;
+
+    /// <content>
+    /// A visitor for the <see cref="RangeAggregationExpression"/> element.
+    /// </content>
+    internal partial class ElasticSearchDSLVisitor : IVisitor
+    {
+        /// <inheritdoc/>
+        public void Visit(DateRangeAggregationExpression dateRangeExpression)
+        {
+            Ensure.IsNotNull(dateRangeExpression, nameof(dateRangeExpression));
+            EnsureClause.StringIsNotNullOrEmpty(dateRangeExpression.Field, nameof(dateRangeExpression.Field));
+
+            var fromExpr = dateRangeExpression.From != null ? DateMathParser.ParseDateMath(dateRangeExpression.From) : "''";
+            var toExpr = dateRangeExpression.To != null ? DateMathParser.ParseDateMath(dateRangeExpression.To) : "''";
+
+            var gteExpr = dateRangeExpression.From != null ? $"['{dateRangeExpression.Field}'] >= {fromExpr}" : null;
+            var ltExpr = dateRangeExpression.To != null ? $"['{dateRangeExpression.Field}'] < {toExpr}" : null;
+
+            dateRangeExpression.BucketNameKustoQL = $"strcat({fromExpr}, '_', {toExpr})";
```

Comment:
Is this consistent with the naming of the normal range buckets?
Also, why strcat in kusto and not in c#?

Reply:
The `strcat()` is on the Kusto side because the datetime is also generated there, using `make_datetime()`.
I am using underscore '_' to separate the bucket name from/to, since the dash '-' is used in the datetime format.

Reply:
Got it.

I think we should have some sort of unified interface.
Between this, numeric range, and percentiles, each one uses a different separator.

Reply:
Right, I see that we now have `AggregationsConstants.MetadataSeparator` (currently set to '%'), I suppose it makes sense to use that.

Reply:
Fixed to use MetadataSeparator.